annotation-web/src/main/java/eu/europeana/annotation/web/service/controller/jsonld/BaseJsonldRest.java

```diff
@@ -70,7 +63,7 @@ protected ResponseEntity<String> storeAnnotation(MotivationTypes motivation, boo
 	    String annotation, Authentication authentication) throws HttpException {
 	try {
 	    // 0. annotation id
-	    AnnotationId annoId = buildAnnotationId(null);
+	    long annoIdentifier = mongoPersistance.generateAnnotationIdentifier();
```

Comment:
no id must be generated before validation, the id must be generated in the store annotations

Reply:
this is probably called both by create and update methods, the create method must not set and Id and the update must have an id